### ISSUE TITLE
Open sample menu data is missing

### ISSUE THREAD

**Issue by sofroniewn** [napari/napari#3968]
## 🐛 Bug

The open sample menu data is missing for me. @tlambert03 can you check if it is there for you too. I just upgraded npe2 / skimage etc. This could be a regression for `0.4.13` that we want to patch right away. Thanks!!

**Comment by tlambert03** [comment]
it's there for me, both in my dev environment and a clean one

**Comment by sofroniewn** [comment]
> it's there for me, both in my dev environment and a clean one

that's a relief - i'll poke around, see if i can make it come back without nuking my env. any guesses what might be going on?

**Comment by sofroniewn** [comment]
i'm on `3.7` ....

**Comment by tlambert03** [comment]
still there for me.  can you test a new env?  (that's all I'm doing here...)

**Comment by sofroniewn** [comment]
ok, will do, i'm making some vids for tweet series. might take me a bit, but thanks!!

**Comment by lauramarcos** [comment]
It is not working for me either in MAC and Windows

MAC:
napari: 0.4.13
Platform: macOS-11.6.2-x86_64-i386-64bit
System: MacOS 11.6.2
Python: 3.9.6 (default, Jul 31 2021, 05:03:56) [Clang 12.0.5 (clang-1205.0.22.9)]
Qt: 5.15.2
PySide2: 5.15.2
NumPy: 1.19.3
SciPy: 1.7.3
Dask: 2022.01.0
VisPy: 0.9.4

OpenGL:
- GL version: 2.1 ATI-4.6.21
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1792x1120, scale 2.0

WIN:

napari: 0.4.13
Platform: Windows-10-10.0.17763-SP0
Python: 3.8.3 (tags/v3.8.3:6f8c832, May 13 2020, 22:37:02) [MSC v.1924 64 bit (AMD64)]
Qt: 5.15.2
PySide2: 5.15.2
NumPy: 1.19.3
SciPy: 1.7.3
Dask: 2022.01.0
VisPy: 0.9.4

OpenGL:
- GL version: 4.5.0 NVIDIA 370.41
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1691x989, scale 1.0


https://user-images.githubusercontent.com/63799148/149988482-1fd40bc9-0e89-49ec-8ae1-d55d3909e488.mov

**Comment by tlambert03** [comment]
@lauramarcos, do you definitely have scikit-image installed too?

PS. lovely image :)

**Comment by tlambert03** [comment]
and @sofroniewn, were you using pyside?

**Comment by lauramarcos** [comment]
@tlambert03 yeah.. according to the napari help's info I do have it install it in both

WIN
Plugins:
- affinder: 0.2.0
- brainreg: 0.1.2
- btrack-reader: 0.1.0
- console: 0.0.4
- napari-demo: 0.0.2
- ndtiffs: 0.1.2
**- scikit-image**
- svg: 0.1.6

MAC
Plugins:
- brainreg: 0.2.3
- brainreg_standard: 0.2.3
- console: 0.0.4
- ndtiffs: 0.1.2
**- scikit-image**

**Comment by tlambert03** [comment]
huh, I'm really struggling to reproduce this one.  every new environment I try seems to work, pyqt5, pyside2, py37, py39, etc...  will keep digging

**Comment by psobolewskiPhD** [comment]
What version of scikit-image? I had something similar happen. I think updating scikit-image fixed it.
Edit: perhaps try installing a plugin that populates that menu? Like pyclesperanto-assistant—maybe a bit heavy for testing...
Wait, LOL, you could use this one from the hackathon:
https://github.com/jdeschamps/napari-sklearn-decomposition
🤣

**Comment by sofroniewn** [comment]
I'm on main and see

```
napari: 0.4.11.dev110+gb43e72aaa
Platform: Darwin-19.6.0-x86_64-i386-64bit
System: MacOS 10.15.7
Python: 3.7.4 (default, Aug 13 2019, 15:17:50) [Clang 4.0.1 (tags/RELEASE_401/final)]
Qt: 5.14.1
PyQt5: 5.14.1
NumPy: 1.21.1
SciPy: 1.7.1
Dask: 2021.07.2
VisPy: 0.9.4
```

not sure why 0.4.11dev ... i've also got latest npe2 /skimage. i will try clean env

**Comment by tlambert03** [comment]
> not sure why 0.4.11dev

prolly need to `git fetch --tags`  (since version is based on the tags in your local git branch the last time you ran pip install... it things you're ~110 commits past the last tag, version 0.4.11)

**Comment by sofroniewn** [comment]
when i try that i get
```
From https://github.com/napari/napari
 ! [rejected]            latest     -> latest  (would clobber existing tag)
```
i dunno my env maybe borked

**Comment by tlambert03** [comment]
that's because of the nightly "latest" tag workflow that got set up:

https://github.com/napari/napari/blob/f8a460d561e109bc77d9d4bba72c890eda1f02ae/.github/workflows/make_bundle.yml#L99-L106

not really a big fan of that approach.  So, you need to run `git fetch --tags --force` to bring your tags back in line with remote

**Comment by tlambert03** [comment]
@sofroniewn and @lauramarcos, when you get a moment, can you both give me your full `pip list`

**Comment by tlambert03** [comment]
I just ran the following code with multiple python/qt combinations, on two mac computers (12.1 and 10.15.7), and still haven't found a combination that repeats the problem.  Please do confirm that you've both tried new environments:

```sh
conda create -n t -c conda-forge python=3.10  # or whatever
conda activate t
pip install "napari[pyqt5]" scikit-image  # or [pyside2]
napari
```

**Comment by lauramarcos** [comment]
I didnt use code to install napari but the installers... also, i am using Mac version Big sur (11.6.2), how can i get you the pip list? @tlambert03

**Comment by tlambert03** [comment]
ah, ok... let's not worry about that for the moment.  let's see if we can determine the problem outside of the bundle before addressing the bundle

**Comment by tlambert03** [comment]
actually... maybe scratch that :joy: I _can_ reproduce in the bundle... and it will likely help figure it out :)

**Comment by tlambert03** [comment]
ok, in the bundle I see what the problem is (not sure this is what @sofroniewn is seeing, but we can check):

In the bundle, if you open the napari console and run
```py
from importlib.metadata import distribution

print(distribution('napari').entry_points)
```
you'll get an empty list (it's not finding any entry points for the bundled napari library).  indeed, there's no `entry_points.txt` file in `napari/Contents/Resources/app/napari-0.4.13-dist-info` where it should be.  (need to go back and confirm that it's been there for previous versions).  so the bundle is currently very broken because it can't find any of the builtins... so it seems to be more of an issue in how napari gets "installed" into the bundle than anything (since the manifest is there).  All the other packages in `app_packages` however (like `napari-svg`) are installed fine and have their entry_points intact

@jaimergp ... not sure whether you'll have any thoughts here, but pinging just in case.

**Comment by tlambert03** [comment]
@sofroniewn can you run the code above and see what you get?  (but first, try a fresh environment... if that fixes it, don't bother)

note, if you're on python 3.7 you'll need:
```py
from importlib_metadata import distribution

print(distribution('napari').entry_points)
```

and then if nothing, print out `sys.path` ... and then look in your repo's `napari.egg-info` folder (assuming you installed with `pip install -e .`) and check for the `entry_points.txt` file

**Comment by jaimergp** [comment]
I check the logs for the bundle building and can't see anything obviously wrong in there :/

**Comment by tlambert03** [comment]
thanks for looking!

**Comment by jaimergp** [comment]
The briefcase versions (the app itself, but also the templates and support packages) haven't changed for a few weeks, so that's not it either...?

**Comment by tlambert03** [comment]
ok, I went back quite a ways and it looks like briefcase has never included the entry_points of the package it's installing... which is unfortunate :/

I guess we'll have to special case this in the bundle process

**Comment by tlambert03** [comment]
@lauramarcos, your problem should be in the nightly build after #3982 merges ... @sofroniewn, not sure about yours, lemme know about the new env testing

**Comment by jaimergp** [comment]
FYI, conda bundles for post 0.4.13 do not exhibit this issue:

![image](https://user-images.githubusercontent.com/2559438/150124419-22cd23f1-87a6-4471-a77e-8118216673b0.png)

```
napari: 0.4.14.dev73+g9c4a4401
Platform: Windows-10-10.0.17763-SP0
Python: 3.8.12 | packaged by conda-forge | (default, Oct 12 2021, 21:22:46) [MSC v.1916 64 bit (AMD64)]
Qt: 5.12.9
PyQt5: 5.12.3
NumPy: 1.22.0
SciPy: 1.7.3
Dask: 2022.01.0
VisPy: 0.9.4

OpenGL:
- GL version: 4.5.0 NVIDIA 370.41
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1440x900, scale 1.0

Plugins:
- console: 0.0.4
- scikit-image: 0.4.14.dev73+g9c4a4401
- svg: 0.1.5
```

No clue why the `scikit-image` plugin version is the same as napari there though?

**Comment by tlambert03** [comment]
That's good, and I'm not surprised.  The lack of the entry_points file in the briefcase bundle is very strange 🤔

Don't worry about scikit image, that's a (confusing) idiosyncrasy of the fact that napari provides that plugin itself

**Comment by chili-chiu** [comment]
I can reproduce the same issue (sample menu data missing) with bundled app installation on mac with napari: 0.4.13.

I tried the nightly build (napari: 0.4.14.dev10+ga699309f) which does bring the sample menu data back, but when clicking on Cells (3D+2Ch), it produces the following error:

ModuleNotFoundError: The requested file is part of the scikit-image distribution, but requires the installation of an optional dependency, pooch. To install pooch, use your preferred python package manager. Follow installation instruction found at https://scikit-image.org/docs/stable/install.html

Some other images can be opened fine.

**Comment by tlambert03** [comment]
Sounds like we just need to add pooch to the bundle deps

**Comment by sofroniewn** [comment]
This is now working for me - same env - not sure what changed tbh. we should add pooch to bundle deps. i can make a new issue for that. i will close this one
